docs/site/Loopback-component-authorization.md

```diff
@@ -8,543 +8,377 @@ permalink: /doc/en/lb4/Loopback-component-authorization.html
-  // .....
-  // other attributes here
-  // .....
-
-  @property.array(String, {
-    required: true,
-  })
-  permissions: PermissionKey[];
+  async function compareId(
+    authorizationCtx: AuthorizationContext,
+    metadata: MyAuthorizationMetadata,
+  ) {
+    let currentUser: UserProfile;
+    if (authorizationCtx.principals.length > 0) {
+      const user = _.pick(authorizationCtx.principals[0], [
+        'id',
+        'name',
+        'email',
+      ]);
+      return AuthorizationDecision.ALLOW;
+    } else {
+      return AuthorizationDecision.DENY;
+    }
+  }
 
-  constructor(data?: Partial<Role>) {
-    super(data);
+  @authenticate('jwt')
+  @authorize({resource: 'order', scopes: ['patch'], voters: [compareId]})
+  async patchOrders(
+    @param.path.string('userId') userId: string,
+    @requestBody() order: Partial<Order>,
+    @param.query.string('where') where?: Where<Order>,
+  ): Promise<Count> {
+    return this.userRepo.orders(userId).patch(order, where);
   }
-}
 ```
 
-- Add user level permissions array to the user model. Do this if there is a use
-  case of explicit allow/deny of permissions at user-level in the application.
-  You can skip otherwise.
-
-{% include code-caption.html content="/src/models/user.model.ts" %}
+In the above example `compareId()` is a authorizing function which is provided
+as a voter in the decorator for the `patchOrders()` method.
+
+## Authorization by decision matrix
+
+The final decision to allow access for a subject is done by the interceptor by
+creating a decision matrix from the voting results (from all the `authorizer` or
+`voter` functions of an endpoint).
+
+The following table illustrates an example decision matrix with 3 votes and
+corresponding options.
+
+| Vote #1 | Vote # 2 | Vote #3 | Options                  | Final Decision |
+| ------- | -------- | ------- | ------------------------ | -------------- |
+| Deny    | Deny     | Deny    | **any**                  | Deny           |
+| Allow   | Allow    | Allow   | **any**                  | Allow          |
+| Abstain | Allow    | Abstain | **any**                  | Allow          |
+| Abstain | Deny     | Abstain | **any**                  | Deny           |
+| Deny    | Allow    | Abstain | {precedence: Deny}       | Deny           |
+| Deny    | Allow    | Abstain | {precedence: Allow}      | Allow          |
+| Allow   | Abstain  | Deny    | {precedence: Deny}       | Deny           |
+| Allow   | Abstain  | Deny    | {precedence: Allow}      | Allow          |
+| Abstain | Abstain  | Abstain | {defaultDecision: Deny}  | Deny           |
+| Abstain | Abstain  | Abstain | {defaultDecision: Allow} | Allow          |
+
+- Here, if suppose there is an `authorizer` function and 2 voters for an
+  endpoint.
+  - if the `authorizer` function returns `ALLOW`, but voter 1 in authorize
+    decorator returns `ABSTAIN` and voter 2 in decorator returns `DENY`.
+  - In this case, if the options provided while
```

Comment:
Perhaps tell user which row of the table you are talking about; to make it clear.

Also, am I assuming the column data correctly?

![image](https://user-images.githubusercontent.com/6864736/70178116-aa42a580-16a9-11ea-96c1-2905c2ce2c0c.png)

When you say

![image](https://user-images.githubusercontent.com/6864736/70178385-3228af80-16aa-11ea-8291-cf23a1b3d228.png)

I am assuming we are talking about this row

![image](https://user-images.githubusercontent.com/6864736/70178276-f988d600-16a9-11ea-9d6d-72c02d0cfd84.png)
 
am I correct?

Reply:
@emonddr perfect ! yes, that is the row I am referring to. I will change the columns names as per your suggestion, to make this more clear.